core-java-modules/core-java-optional/src/test/java/com/baeldung/optionaluses/UsesForOptionalUnitTest.java

```diff
@@ -42,23 +43,49 @@ public void givenExistentUserId_whenFoundUserWithNameStartingWithMInRepositoryUs
 
         if (user != null) {
             if (user.getName().startsWith("M")) {
-                upperCasedName = user.getName().toUpperCase();
+                upperCasedName = user.getName()
+                    .toUpperCase();
             }
         }
 
         assertEquals("MARIA", upperCasedName);
     }
 
     @Test
-    public void givenExistentUserId_whenFoundUserWithNameStartingWithMInRepositoryUsingOptional_thenNameShouldBeUpperCased() {
+    void givenExistentUserId_whenFoundUserWithNameStartingWithMInRepositoryUsingOptional_thenNameShouldBeUpperCased() {
 
         UserRepositoryWithOptional userRepositoryWithOptional = new UserRepositoryWithOptional();
 
         String upperCasedName = userRepositoryWithOptional.findById("2")
-          .filter(u -> u.getName().startsWith("M"))
-          .map(u -> u.getName().toUpperCase())
-          .orElse("");
+            .filter(u -> u.getName().startsWith("M"))
+            .map(u -> u.getName().toUpperCase())
+            .orElse("");
 
         assertEquals("MARIA", upperCasedName);
     }
+
+    @Test
+    void givenExistentUserId_whenSearchForUser_thenThrowException() {
+
+        final UserRepositoryWithOptional userRepositoryWithOptional = new UserRepositoryWithOptional();
+        String existentUserId = "2";
+
+        assertThrows(UserFoundException.class, () -> {
+            userRepositoryWithOptional.throwExceptionWhenUserIsPresent(existentUserId);
+        });
+
+    }
+
+    @Test
+    void givenNonExistentUserId_whenSearchForUser_thenDoNotThrowException() {
+
+        final UserRepositoryWithOptional userRepositoryWithOptional = new UserRepositoryWithOptional();
+        String nonExistentUserId = "8";
+
+        assertDoesNotThrow(() -> {
+            userRepositoryWithOptional.throwExceptionWhenUserIsPresent(nonExistentUserId);
+        });
+
+    }
```

Comment:
You can remove parentheses to convert from lambda statement to expression lambda:

`        assertDoesNotThrow(() -> userRepositoryWithOptional.throwExceptionWhenUserIsPresent(nonExistentUserId));`

Reply:
Done